packages/mobile/src/account/LinkBankAccountScreen.tsx

```diff
@@ -22,92 +26,159 @@ import VerificationPending from 'src/icons/VerificationPending'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { linkBankAccountStepTwoEnabledSelector } from '../app/selectors'
+import { fetchFinclusiveKyc } from './actions'
 import openPlaid, { handleOnEvent } from './openPlaid'
 
-interface StepOneProps {
-  kycStatus: KycStatus | undefined
-}
-
 function LinkBankAccountScreen() {
-  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
   const navigation = useNavigation()
-  React.useEffect(() => {
+
+  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
+  useEffect(() => {
     const unsubscribe = navigation.addListener('beforeRemove', (e) => {
       ValoraAnalytics.track(CICOEvents.link_bank_account_cancel)
     })
     // Unsubscribe will be called on unmount
     return unsubscribe
   }, [])
 
-  const kycStatus = useSelector(kycStatusSelector)
-  const stepTwoEnabled = useSelector(linkBankAccountStepTwoEnabledSelector)
-
   return (
     <SafeAreaView style={styles.body}>
       <ScrollView
         contentContainerStyle={styles.contentContainer}
         showsVerticalScrollIndicator={false}
       >
-        <StepOne kycStatus={kycStatus} />
-        <StepTwo disabled={!stepTwoEnabled || kycStatus !== KycStatus.Approved} />
+        <StepOne />
+        <StepTwo />
       </ScrollView>
     </SafeAreaView>
   )
 }
 
-export function StepOne({ kycStatus }: StepOneProps) {
+interface StepOneUIStateParams {
+  kycStatus: KycStatus | undefined
+  finclusiveKycStatus: FinclusiveKycStatus
+  errorFromPersona: boolean
+  successFromPersona: boolean
+  isInPersonaFlow: boolean
+}
+
+enum StepOneUIState {
+  Begin = 'Begin',
+  Failure = 'Failure',
+  Pending = 'Pending',
+  Completed = 'Completed',
```

Comment:
Took me a sec to realize that 'Completed' is the terminal state here, whereas with Persona 'Completed' actually means it's still pending; I like this convention a lot better though, much cleaner imo.

Reply:
I know, i hate all the different state names 😖